### DISCUSSION TITLE
Controlling the code gen java package names

### DISCUSSION THREAD

**Discussion by zzur** [looker-open-source/sdk-codegen#513] [Q&A]
Following the README, it was suggested to use the _legacy_ code generator for java. The following section in the _codeGenerators.js_ file suggestion there is an option to control the generated java code's package name. 

` {
    language: 'java',
    legacy: 'java',
    options: '-papiPackage=foo.bar.client.api.v3.1 -ppackageName=foo.bar.client.api.v3.1',
    extension: /\.java/gi,
  },
`

I was not able to get this to work. Any insights on the two properties in the options (the difference between the two?) and if they are expected to work?

**Comment by zzur** [comment]
was trying to poke around to fine more in the code, I may be wrong about the meaning of these properties...
Is there any option to control the java packages in the generated code?

**Comment by jkaster** [comment]
the options argument for the package names is what was correct at the time I tested the `legacy` feature for Java generation. The names may have changed in an update to that generator. I haven't checked on Java generation support since the Kotlin SDK is now available for Java/JVM usage.
